python/lsst/ip/isr/deferredCharge.py

```diff
@@ -378,7 +383,25 @@ def fromTable(cls, tableList):
             ampTrap['emissionTime'] = emissionTimes[index]
             ampTrap['pixel'] = pixels[index]
             ampTrap['trap_type'] = trap_type[index]
-            ampTrap['coeffs'] = np.array(coeffs[index])[~np.isnan(coeffs[index])].tolist()
+
+            # Unpad any trailing NaN values: find the continuous array
+            # of NaNs at the end of the coefficients, and remove them.
+            inCoeffs = coeffs[index]
+            breakIndex = 1
+            nanValues = np.where(np.isnan(inCoeffs))[0]
+            if nanValues is not None:
+                coeffLength = len(inCoeffs)
+                while (breakIndex < coeffLength):
+                    if coeffLength - breakIndex in nanValues:
+                        breakIndex += 1
+                    else:
+                        break
+            breakIndex -= 1  # Remove the fixed offset.
+            if breakIndex != 0:
+                outCoeffs = inCoeffs[0:coeffLength - breakIndex]
```

Comment:
You might need a space either side of the `:` here.

Reply:
Flake8 didn't like the leading space, so I only added the one.

Reply:
black will add the space but this package doesn't use black (which requires a tweak to flake8 settings).